legion/sdk/legion/sdk/clients/edi.py

```diff
@@ -103,27 +170,37 @@ def construct_from_other(cls, other):
         """
         return cls(other._base, other._token, other._retries, other._timeout)
 
-    def _request(self, url_template: str, payload: typing.Mapping[typing.Any, typing.Any] = None, action: str = 'GET',
-                 stream: bool = False, timeout: typing.Optional[int] = None):
+    def _request(self,
+                 url_template: str,
+                 payload: typing.Mapping[typing.Any, typing.Any] = None,
+                 action: str = 'GET',
+                 stream: bool = False,
+                 timeout: typing.Optional[int] = None,
+                 limit_stack: bool = False):
         """
         Make HTTP request
+
+        :param url_template: target URL
+        :type url_template: str
+        :param payload: (Optional) data to be placed in body of request
+        :type payload: dict[str, str] or None
         :param action: request action, e.g. get / post / delete
         :type action: str
-        :param url: target URL
-        :type url: str
-        :param data: (Optional) data to be placed in body of request
-        :type data: dict[str, str] or None
-        :param headers: (Optional) additional HTTP headers
-        :type headers: dict[str, str] or None
-        :param cookies: (Optional) HTTP cookies
-        :type cookies: dict[str, str] or None
+        :param stream: (Optional) use stream mode or not
+        :type stream: bool
         :param timeout: (Optional) custom timeout in seconds (overrides default). 0 for disabling
         :type timeout: typing.Optional[int]
+        :param limit_stack: (Optional) do not start refreshing token if it is possible
+        :type limit_stack: bool
         :return: :py:class:`requests.Response` -- response
         """
         sub_url = url_template.format(version=self._version)
         target_url = self._base.strip('/') + sub_url
-        cookies = {'_oauth2_proxy': self._token} if self._token else {}
+        headers = {}
+
+        # Add token if it is provided
+        if self._token:
```

Comment:
When I pass a JWT token from an EDI server explicitly, for example `legionctl login --edi ... --token ...`, I still redirect to Keycloak.
The first problem, I get a redirect response if the token is correct.
The second problem,  I get a redirect response if the token is wrong. IMHO, if I pass token explicitly, I expect the login command will fail if the token is incorrect.

Reply:
You can't get redirect if token is correct. Could you please reproduce it for me?
Second problem is a fact use-case specific and I think we should not place this logic in SDK.

Reply:
1. I copy token from the edi web page. And then execute command 
```bash
legionctl --verbose login --edi https://edi.gke-legion-test.ailifecycle.org --token "${TOKEN}"
``` 
```
curl -H "Authorizarion: Bearer ${TOKEN}" https://edi.gke-legion-test.ailifecycle.org -I
>HTTP/2 302 
>server: nginx/1.13.12
```
Maybe JWT in web page is not correct?
2. The main point is I do not understand why we need the non-interactive flag? Let's get rid of it. We have two situations:
*  When a user provides token in client or CLI explicitly then we do not use redirect and fail if the token is wrong.
* When a user uses client or CLI without token then we authenticate using a browser.

Reply:
I'l implement it in my PR